pkg/user/apis/user/v1/conversion.go

```diff
@@ -2,17 +2,49 @@ package v1
 
 import (
 	"k8s.io/apimachinery/pkg/runtime"
-
-	"github.com/openshift/origin/pkg/api/apihelpers"
-	userapi "github.com/openshift/origin/pkg/user/apis/user"
 )
 
 func addConversionFuncs(scheme *runtime.Scheme) error {
-	if err := scheme.AddFieldLabelConversionFunc("v1", "Identity",
-		apihelpers.GetFieldLabelConversionFunc(userapi.IdentityToSelectableFields(&userapi.Identity{}), nil),
-	); err != nil {
+	return nil
+}
+
+func addLegacyFieldSelectorKeyConversions(scheme *runtime.Scheme) error {
+	if err := scheme.AddFieldLabelConversionFunc(LegacySchemeGroupVersion.String(), "Identity", legacyIdentityFieldSelectorKeyConversionFunc); err != nil {
 		return err
 	}
+	return nil
+}
 
+func addFieldSelectorKeyConversions(scheme *runtime.Scheme) error {
+	if err := scheme.AddFieldLabelConversionFunc(SchemeGroupVersion.String(), "Identity", identityFieldSelectorKeyConversionFunc); err != nil {
+		return err
+	}
 	return nil
 }
+
+// because field selectors can vary in support by version they are exposed under, we have one function for each
+// groupVersion we're registering for
+
+func legacyIdentityFieldSelectorKeyConversionFunc(label, value string) (internalLabel, internalValue string, err error) {
+	switch label {
+	case "providerName",
+		"providerUserName",
+		"user.name",
+		"user.uid":
+		return label, value, nil
+	default:
+		return runtime.DefaultMetaV1FieldSelectorConversion(label, value)
```

Comment:
`return apihelpers.LegacyMetaV1FieldSelectorConversionWithName(label, value)`?

Reply:
> return apihelpers.LegacyMetaV1FieldSelectorConversionWithName(label, value)?

This one never had the "name" -> "metadata.name" mapping.